### PR TITLE
ORV2-1496 Change Payment gateway From MotiPay to PayBC

### PR DESCRIPTION
**Pull request by gchauhan-aot**
# Description

This PR has changes for payment gateway. MotiPay is replaced by PayBC. FE redirect url has been changed. hash validation for PayBC response has been added.
Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Manually tested using Swagger.
- [x] Integration testing with Frontend.

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have already been accepted and merged

## Further comments

---

Thanks for the PR!

Any successful deployments (not always required) will be available below.
[Backend - Vehicles](https://onroutebc-696-backend-vehicles.apps.silver.devops.gov.bc.ca/) available
[Backend - DOPS](https://onroutebc-696-backend-dops.apps.silver.devops.gov.bc.ca/) available
[Frontend](https://onroutebc-696-frontend.apps.silver.devops.gov.bc.ca/) available

Once merged, code will be promoted and handed off to following workflow run.
[Main Merge Workflow](https://github.com/bcgov/onroutebc/actions/workflows/merge-main.yml)

### CODE REVIEW

File: frontend/src/features/permits/helpers/payment.ts [hunk 1]
```diff
@@ -24,9 +28,9 @@ export const getMotiPaymentDetails = (params: URLSearchParams): MotiPaymentDetai
     avsProcessed: getDefaultRequiredVal("", params.get("avsProcessed")),
     avsResult: getDefaultRequiredVal("", params.get("avsResult")),
     cardType: getDefaultRequiredVal(CARD_TYPES.VI, params.get("cardType")) as CardType,
-    cvdId: applyWhenNotNullable((cvdId) => Number(cvdId), params.get("cvdId"), 0),
-    trnApproved: applyWhenNotNullable((trnApproved) => Number(trnApproved), params.get("trnApproved"), 0),
-    messageId: getDefaultRequiredVal("", params.get("messageId")),
+    cvdId: 1, // applyWhenNotNullable((cvdId) => Number(cvdId), params.get("cvdId"), 0),
+    trnApproved: trnApproved,
+    messageId: '1', // getDefaultRequiredVal("", params.get("messageId")),
```
Comment (erikataot) [inline frontend/src/features/permits/helpers/payment.ts:33]: needed to hardcode both `cvdId` and `messageId` values in order for submissions to work, as they weren't showing up properly in the response (from PayBC?) that we're parsing here. Not sure if these are just "extra" values no longer needed going from MOTI to PayBC

File: backend/vehicles/src/common/helper/validateHash.helper.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import * as CryptoJS from 'crypto-js';
+
+export const validateHash = (
+  queryString: string,
+  hashvalue: string,
+): boolean => {
+  const payBCHash: string = CryptoJS.MD5(
+    `${queryString}${process.env.PAYBC_API_KEY}`,
+  ).toString();
+  if (hashvalue === payBCHash) return true;
```
Comment (erikataot) [inline backend/vehicles/src/common/helper/validateHash.helper.ts]: should be able to restructure this as `return hashvalue === payBCHash`

**Review by erikataot — DISMISSED** [review]
I'm not a registered and licensed `CODEOWNER` but I approve this 🌮

**Review by praju-aot — DISMISSED** [review]
Approved

**Review by krishnan-aot — DISMISSED** [review]

File: backend/vehicles/src/common/helper/validateHash.helper.ts (hunk 2 above)
Comment (gchauhan-aot) [reply backend/vehicles/src/common/helper/validateHash.helper.ts]: done

**Comment by erikataot** [comment]
a future suggestion for work like this (IMHO) would be to split out the OpenShift changes and just add any necessary ENV configs in the first PR on its own. Makes it much smaller. Then when work is merged to actually cutover and use the new ENV, make another small PR for cleanup of the unused. I dunno. Smaller PR's are always easier to get moving then larger ones.

**Review by erikataot — DISMISSED** [review]

**Review by cberg-aot — DISMISSED** [review]

**Review by krishnan-aot — DISMISSED** [review]

**Review by praju-aot — DISMISSED** [review]
approved

**Review by cberg-aot — DISMISSED** [review]

**Review by cberg-aot — APPROVED** [review]

**Review by erikataot — APPROVED** [review]

**Pull request merged**